### ISSUE TITLE
Is not awaitable is incorrectly reported for class instances that define an `__await__` method

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#3453]
**Describe the bug**

```py
import asyncio
from typing import Iterator

class MyAwaitable:
    def __await__(self) -> Iterator[int]:
        async def foo() -> int:
            return 1

        return foo().__await__()

async def main() -> None:
    print(await MyAwaitable())

asyncio.run(main())
```

Pyright reports:

```
error: "MyAwaitable" is not awaitable
```

When this works correctly at runtime:

```
1
```

**Expected behavior**

No errors should be reported and the result of the await call should be inferred to `int`.

**VS Code extension or command-line**

Command-line: 1.1.246

**Comment by erictraut** [comment]
This is because you've provided the wrong return type annotation for the `__await__` method. It should be `Generator[Any, None, int]`. If you make this change, it type checks without errors.
